napari/_qt/containers/qt_axis_model.py

```diff
@@ -71,8 +75,41 @@ def setData(
     ) -> bool:
         axis = self.getItem(index)
         if role == Qt.ItemDataRole.CheckStateRole:
-            axis.set_rollable(Qt.CheckState(value) == Qt.CheckState.Checked)
+            axis.rollable = Qt.CheckState(value) == Qt.CheckState.Checked
+        elif role == Qt.ItemDataRole.EditRole:
+            axis_labels = list(axis.dims.axis_labels)
+            axis_labels[axis.axis] = value
+            axis.dims.axis_labels = axis_labels
         else:
             return super().setData(index, value, role=role)
         self.dataChanged.emit(index, index, [role])
         return True
+
+    def flags(self, index: QModelIndex) -> Qt.ItemFlags:
+            """Returns the item flags for the given `index`.
+
+            This describes the properties of a given item in the model.  We set
+            them to be editable, checkable, dragable, droppable, etc...
+            If index is not a list, we additionally set `Qt.ItemNeverHasChildren`
+            (for optimization). Editable models must return a value containing
+            `Qt.ItemIsEditable`.
+
+            See Qt.ItemFlags https://doc.qt.io/qt-5/qt.html#ItemFlag-enum
+            """
+            flags = (
+                Qt.ItemFlag.ItemIsSelectable
+                | Qt.ItemFlag.ItemIsEditable
+                | Qt.ItemFlag.ItemIsUserCheckable
+                | Qt.ItemFlag.ItemIsTristate
```

Comment:
Do we need tri-state items here? If so, what does partially checked mean?

If we do need them, then use `ItemIsUserTristate` instead of the deprecated `ItemIsTristate` to avoid Qt6 compatibility issues: https://doc.qt.io/qt-5/qt.html#ItemFlag-enum

Reply:
Oh, this is shouldn't be necessary. Just a residual. Will remove this.

Reply:
Are you going to push new changes with the removal up? If so, I'll wait for that, then will merge. Otherwise, I'll just merge ASAP.

Reply:
Just merge, I will clean this in the main pull request.